File: Automation/001-AutomationFramework/C#/AutomationFramework/Common/SharedTestSteps/EmailServices/MailinatorTestSteps.cs
using CommonCore.Tests;
using SeleniumEngine.DriverInitialization;
using SharedTestSteps.Contracts;
using UIMappings.Pages.EmailProviders;

namespace SharedTestSteps.EmailServices
{
    public class MailinatorTestSteps : IEmailServiceOperator
    {
        private string EmailPrefix
        {
            get
            {
                return String.Format("franchiTest{0}", DateTime.UtcNow.ToString("yyMMddhmsff"));
            }
        }

        public void GenerateEmail()
        {
            TestInMemoryParameters.Instance.GeneratedEmailAddress = string.Format("{0}@{1}", EmailPrefix, "mailinator.com");
        }

        public void GetEmailAddress()
        {
            try
            {
                WebDriverFactory.WebDriver.GetPage<Mailinator>()
                                          .EnterEmail(EmailPrefix)
                                          .ClickOnGo()
                                          .ImplicitlyWaitForPageToBeReady<Mailinator>(2)
                                          .SaveGeneratedEmailAddress(TestInMemoryParameters.Instance.SetEmailFor);

                WebDriverFactory.TakeScreenshot("MAILINATOR - Get email address");
            }
            catch (Exception ex)
            {
                throw new Exception("Unable to get generated email address from 'Mailinator'", ex);
            }
        }

        public void GetVerificationCodeFromEmail()
        {
            string verificationCodeText = string.Empty;

            WebDriverFactory.WebDriver.GetPage<Mailinator>()
                                      .EnterEmail(TestInMemoryParameters.Instance.GeneratedEmailAddress)
                                      .ClickOnGo()
                                      .WaitForEmail(TestInMemoryParameters.Instance.VerificationSenderAddress)
                                      .ClickOnEmail(TestInMemoryParameters.Instance.VerificationSenderAddress)
                                      .GetVerificationCode(TestInMemoryParameters.Instance.VerificationCode, out verificationCodeText);

            TestInMemoryParameters.Instance.VerificationCode = verificationCodeText.Split(new string[] { ": " }, StringSplitOptions.None)[1].Substring(0, 6);

            WebDriverFactory.TakeScreenshot("MAILINATOR - Get verification code");
        }

        public void VerifyEmailNotificationContent()
        {
            WebDriverFactory.WebDriver.GetPage<Mailinator>()
                                      .EnterEmail(TestInMemoryParameters.Instance.GeneratedEmailAddress)
                                      .ClickOnGo()
                                      .WaitForEmail(TestInMemoryParameters.Instance.EmailSender)
                                      .ClickOnEmail(TestInMemoryParameters.Instance.EmailSender)
                                      .ImplicitlyWaitForPageToBeReady<Mailinator>(3)
                                      .VerifyEmailBodyContent(TestInMemoryParameters.Instance.EmailText)
                                      .ImplicitlyWaitForPageToBeReady<Mailinator>(2);
        }

        public void DeleteAllEmailsFromInbox()
        {
            WebDriverFactory.WebDriver.GetPage<Mailinator>()
                                      .EnterEmail(TestInMemoryParameters.Instance.GeneratedEmailAddress)
                                      .ClickOnGo()
                                      .SelectAllEmailsFromInbox()
                                      .ClickOnDelete();
        }
    }
}

File: Automation/Woolsocks/TestAutomationFramework/src/test/java/testSuites/shared/steps/emailServices/MailinatorTestSteps.java
package testSuites.shared.steps.emailServices;

import common.core.TestInMemoryParameters;
import common.utilities.LoggingManager;
import testSuites.shared.steps.contracts.EmailServiceOperator;
import uiMappings.pages.emailProviders.Mailinator;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class MailinatorTestSteps extends EmailServiceOperator {
    private String getEmailPrefix() {
        LocalDateTime currentTime = LocalDateTime.now();
        LocalDate date = currentTime.toLocalDate();
        String timeString = String.format("%s_%s_%s", currentTime.getHour(), currentTime.getMinute(), currentTime.getSecond());

        return String.format("TestEmail_%s_%s", date, timeString);
    }

    public void saveGeneratedEmail() {
        String temporaryEmail = String.format("%s@%s", getEmailPrefix(), "mailinator.com");
        TestInMemoryParameters.getInstance().generatedEmailAddress = temporaryEmail;
        LoggingManager.Info("Temporary email created: " + temporaryEmail);
    }

    public void getEmailAddress() {
        try {
            String emailPrefix = getEmailPrefix();

            getPage(Mailinator.class).
                    clikOnPublicInbox().
                    enterEmail(emailPrefix).
                    clickOnGo().
                    ImplicitlyWaitForPageToBeReady(Mailinator.class, 3);

            saveGeneratedEmail();
        } catch (Exception ex) {

        }
    }

    public void verifyEmailNotificationContent() {
        try {
            getPage(Mailinator.class).
                    enterEmail(TestInMemoryParameters.getInstance().generatedEmailAddress).
                    clickOnGo().
                    waitForEmail(TestInMemoryParameters.getInstance().emailSender).
                    clickOnEmail(TestInMemoryParameters.getInstance().emailSender).
                    ImplicitlyWaitForPageToBeReady(Mailinator.class, 3).
                    verifyEmailBodyContent(TestInMemoryParameters.getInstance().emailText).
                    ImplicitlyWaitForPageToBeReady(Mailinator.class, 3);
        }
        catch (Exception exception) { }
    }
}
